lxd/storage/drivers/driver_cephfs.go

```diff
@@ -129,9 +129,74 @@ func (d *cephfs) Create() error {
 		fsPath = fields[1]
 	}
 
-	// Check that the filesystem exists.
-	if !d.fsExists(d.config["cephfs.cluster_name"], d.config["cephfs.user.name"], fsName) {
-		return fmt.Errorf("The requested '%v' CEPHFS doesn't exist", fsName)
+	// If the filesystem already exists, disallow keys associated to creating the filesystem.
+	fsExists, err := d.fsExists(d.config["cephfs.cluster_name"], d.config["cephfs.user.name"], fsName)
+	if err != nil {
+		return fmt.Errorf("Failed to check if %q CEPHFS exists: %w", fsName, err)
+	}
+
+	if fsExists {
+		for _, key := range []string{"create_missing", "osd_pg_num", "meta_pool", "data_pool"} {
+			cephfsSourceKey := fmt.Sprintf("cephfs.%s", key)
+			if d.config[cephfsSourceKey] != "" {
+				return fmt.Errorf("Invalid config key %q: CephFS filesystem already exists", cephfsSourceKey)
```

Comment:
Can we use either CephFS (my preference) or CEPHFS consistently in the error messages please?